src/rpc/core_rpc_server.cpp

```diff
@@ -2582,9 +2582,19 @@ namespace cryptonote
 
     res.status = CORE_RPC_STATUS_OK;
     res.service_node_states.reserve(pubkey_info_list.size());
-    for (const auto &pubkey_info : pubkey_info_list)
+    
+    if (req.include_json)
+    {
+      res.as_json = "{\n}";
```

Comment:
Tried a similar idea, but because the input is a vector type (regardless of being empty), json_archive reads it as an array and prefixes `[` and `]` instead.

Reply:
I see. This is ok then.